### ISSUE TITLE
bad bug in hotspot offset

### ISSUE THREAD

**Issue by rodluger** [rodluger/planetplanet#53]
If I add a latitudinal offset Phi = 30 degrees to a planet in an orbit with inclination = 80 degrees, I get something that looks good:
![image](https://user-images.githubusercontent.com/9323819/29597968-e1bf2210-877a-11e7-853d-a6fa428dc4db.png)

But when the inclination is close to zero, I get the following:
![image](https://user-images.githubusercontent.com/9323819/29593585-025637e0-8760-11e7-9e82-4c1f04a89af8.png)

I think my math in Appendix B.3 may be wrong: specifying an offset in latitude at low inclination leads to an offset in _longitude_, with some other weird results.

To reproduce this bug, run

```
python -c "import planetplanet as pp; import matplotlib.pyplot as pl; pp.DrawOrbit(size = 2, inc = 5, Phi = 30, nz = 11); pl.show()"
```

**Comment by ericagol** [comment]
I think that sounds good.

Eric Agol
Astronomy Professor
University of Washington

**Comment by rodluger** [comment]
OK, I figured it out! Here's the correct solution for an inclination of 5 degrees and a latitudinal offset of 30 degrees:

![image](https://user-images.githubusercontent.com/9323819/29754657-884bce50-8b3e-11e7-9106-eeb01e16938c.png)

The approach using the orbital angles would have worked, but REBOUND doesn't automatically compute the true anomaly, so that would have been slow. I figured out how to use the position and velocity cartesian components to define the latitudinal and longitudinal directions. This is something @ericagol suggested a while back but I didn't quite know how to do.

The idea is to compute the position of the substellar point,

![image](https://user-images.githubusercontent.com/9323819/29754678-f89ac4f4-8b3e-11e7-8a3b-f32db4556a2c.png)

which defines a vector **rstar**. We then use the [Rodrigues formula](https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula) to rotate by **Lambda** along the orbital plane (defined by the normal vector **r x v**), then rotate by **Phi** in a plane defined by the normal vector **(r x v) x v**.

I had to adapt the C code to return the array of velocity vectors, but it's very straightforward. I still need to implement this for the Kepler solver, since I don't know the equations to get the velocity from the orbital elements. I'll figure it out from Murray & Dermott this week.
